packages/core/src/DatumBuilders/DatumBuilder.Lucid.V3.class.ts

```diff
@@ -205,19 +234,91 @@ export class DatumBuilderLucidV3 implements DatumBuilder {
     };
   }
 
-  public buildDepositPair(deposit: TDepositMixed): TDatumResult<Data> {
-    const datum = new Constr(2, [
-      new Constr(1, [
-        new Constr(0, [deposit.CoinAAmount.amount, deposit.CoinBAmount.amount]),
-      ]),
-    ]);
+  /**
+   * Creates a new pool datum for minting a the pool. This is attached to the assets that are sent
+   * to the pool minting contract. See {@link Lucid.TxBuilderLucidV3} for more details.
+   *
+   * @param {IDatumBuilderMintPoolV3Args} params The arguments for building a pool mint datum.
+   *  - assetA: The amount and metadata of assetA. This is a bit misleading because the assets are lexicographically ordered anyway.
+   *  - assetB: The amount and metadata of assetB. This is a bit misleading because the assets are lexicographically ordered anyway.
+   *  - feeDecay: A tuple of bigints, [bigint, bigint], representing the fee at marketOpen, decaying linearly till feeDecayEnd.
+   *  - feeDecayEnd: The POSIX timestamp for when the fee should stop decaying.
+   *  - marketOpen: The POSIX timestamp for when pool trades should start executing.
+   *  - protocolFee: The fee gathered for the protocol treasury.
+   *  - seedUtxo: The UTXO to use as the seed, which generates asset names and the pool ident.
+   *
+   * @returns {TDatumResult<V3Types.TPoolDatum>} An object containing the hash of the inline datum, the inline datum itself,
+   *                                              and the schema of the original pool mint datum, crucial for the execution
+   *                                              of the minting pool operation.
+   */
+  public buildMintPoolDatum({
+    assetA,
+    assetB,
+    feeDecay,
+    feeDecayEnd,
+    marketOpen,
+    protocolFee,
+    seedUtxo,
+  }: IDatumBuilderMintPoolV3Args): TDatumResult<V3Types.TPoolDatum> {
+    const ident = DatumBuilderLucidV3.computePoolId(seedUtxo);
+    const liquidity = sqrt(assetA.amount * assetB.amount);
+
+    const assetsPair = this.buildLexicographicalAssetsDatum(
+      assetA,
+      assetB
+    ).schema;
+
+    const newPoolDatum: V3Types.TPoolDatum = {
+      assets: assetsPair,
+      circulatingLp: liquidity,
+      feesPer10Thousand: feeDecay,
```

Comment:
is there any specific reason why we need to pass the fees in that "per 10 thousand" format?

Reply:
It's per the protocol rules ;)